### ISSUE TITLE
openfoam module issue

### ISSUE THREAD

**Issue by Nandinihpc** [LLNL/spack#22774]
I have installed openfoam2012 successfully. When i did spack module lmod refresh it gives this warning and hangs 

==> Do you want to proceed? [y/n] y
==> Regenerating lmod module files
==> Warning: Quotes in command arguments can confuse scripts like configure.
  The following arguments may cause problems when executed:
      source /dev/null &> /dev/null && python3 -c "import os, json; print(json.dumps(dict(os.environ)))"
  Quotes aren't needed because spack doesn't use a shell.
  Consider removing them
==> Warning: Quotes in command arguments can confuse scripts like configure.
  The following arguments may cause problems when executed:
      source /opt/apps/nfs/spack-quanah/spack/opt/spack/linux-centos8-broadwell/gcc-10.2.0/openfoam-2012-sm3m4wiv32m6t3f5lm43rgvibgvqkruz/etc/bashrc &> /dev/null && python3 -c "import os, json; print(json.dumps(dict(os.environ)))"
  Quotes aren't needed because spack doesn't use a shell.
  Consider removing them
Let me know the fix please.

Thanks

**Comment by Nandinihpc** [comment]
This is the output of loading the openfoam module

 spack --stacktrace load openfoam
lib/spack/spack/util/executable.py:175 ==> Warning: Quotes in command arguments can confuse scripts like configure.
  The following arguments may cause problems when executed:
      source /dev/null &> /dev/null && python3 -c "import os, json; print(json.dumps(dict(os.environ)))"
  Quotes aren't needed because spack doesn't use a shell.
  Consider removing them
lib/spack/spack/util/executable.py:175 ==> Warning: Quotes in command arguments can confuse scripts like configure.
  The following arguments may cause problems when executed:
      source /opt/apps/nfs/spack-quanah/spack/opt/spack/linux-centos8-broadwell/gcc-10.2.0/openfoam-2012-sm3m4wiv32m6t3f5lm43rgvibgvqkruz/etc/bashrc &> /dev/null && python3 -c "import os, json; print(json.dumps(dict(os.environ)))"
  Quotes aren't needed because spack doesn't use a shell.
  Consider removing them

**Comment by Nandinihpc** [comment]
-bash: lib/spack/spack/user_environment.py:95: No such file or directory

**Comment by adamjstewart** [comment]
This is related to #22775 and #22766, can you see if this is fixed by #22767?

**Comment by Nandinihpc** [comment]
Can you explain which part i need to update in order to fix the issue.

**Comment by adamjstewart** [comment]
#22767 just disables the warning message. If there's another issue with the package this won't help. Maybe @olesenm can comment.

**Comment by Nandinihpc** [comment]
How to disable the warning message. I am sorry I am new to spack and I dont have idea on how to disable this.

**Comment by adamjstewart** [comment]
Once #22767 is merged it will be disabled. If you want you can test out the PR now: https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/checking-out-pull-requests-locally

**Comment by Nandinihpc** [comment]
thank you. Applying the fix in executable.py and environment.py clears the warning message.
